contracts/LockedTokens.cdc

```diff
@@ -591,6 +588,51 @@ pub contract LockedTokens {
         }
     }
 
+    /// Provides access to the locked vault in the token manager resource 
+    /// to other contracts that are deployed to the same account as the locked tokens contract
+    /// Enables the StakingCollection to access the locked vault without the user being able to access it
+    pub resource LockedVaultHolder {
+        // Capability to the vault in the locked account
+        access(self) var lockedVault: Capability<&FlowToken.Vault>?
+
+        init() {
+            self.lockedVault = nil
+        }
+
+        access(account) fun getVaultBalance(): UFix64 {
+            let vaultRef = self.lockedVault!.borrow()!
+
+            return vaultRef.balance
+        }
+
+        // A contract in the same account can call these functions when given a LockedVaultHolder,
+        // but nobody else can
+        
+        access(account) fun addVault(lockedVault: Capability<&FlowToken.Vault>) {
```

Comment:
Maybe this should be called `setVault`? "Add" sounds non-destructive, whereas this function replaces the potentially existing locked vault capability